### PR TITLE
Add Api Version differences

### PR DESCRIPTION
**Pull request by PeterChen13579**
Documenting issue [4545](https://github.com/XRPLF/rippled/issues/4545)

### CODE REVIEW

**Comment by PeterChen13579** [comment]
Hi @mDuo13! There was a bunch of PR's that got merged ([here](https://github.com/XRPLF/rippled/issues/4303), [here](https://github.com/XRPLF/rippled/issues/4539) and [here](https://github.com/XRPLF/rippled/issues/4288) regarding bugs/discrepancies for certain user input calls. In general, all the fix(new behavior) is set for api_version 2, while the old behavior is set for api_version 1. I was wondering how we should document this in the docs? Take a look at my changes as an example. (doubt it will actually be like this though) Thanks for the help 😄

**Comment by intelliot** [comment]
For simplicity, I suggest just documenting the latest version (and putting `"api_version": 2` in the request). Of course, this should only be merged/deployed after the new version has been widely deployed in production. There should also an "added in" badge to show which version of `rippled` is being documented.

There can be a small note to indicate that older API versions exist, perhaps with a link to https://github.com/XRPLF/rippled/blob/develop/API-CHANGELOG.md for those who want to read about those historical versions.

**Comment by intelliot** [comment]
"here" is not good [anchor text](https://moz.com/learn/seo/anchor-text). @thejohnfreeman or @mDuo13 , could you perhaps explain why, and offer an alternative?

File: content/references/http-websocket-apis/public-api-methods/account-methods/account_info.md [hunk 1]
```diff
@@ -256,6 +256,11 @@ Each object in the `transactions` array of `queue_data`, if present, may contain
 * `actNotFound` - The address specified in the `account` field of the request does not correspond to an account in the ledger.
 * `lgrNotFound` - The ledger specified by the `ledger_hash` or `ledger_index` does not exist, or it does exist but the server does not have it.
 
+## Api Version 2
+* - Specifying invalid signer_lists value returns invalidParams error
+
+Older Api versions can be used. Documented [here][api-version-history]
```
Comment (thejohnfreeman) [inline content/references/http-websocket-apis/public-api-methods/account-methods/account_info.md]:
```suggestion
Older [API versions][api-version-history] can be used.
```

File: content/references/http-websocket-apis/public-api-methods/account-methods/account_tx.md [hunk 2]
```diff
@@ -460,6 +460,11 @@ Each transaction object includes the following fields, depending on whether it w
 * `lgrIdxMalformed` - The ledger specified by the `ledger_index_min` or `ledger_index_max` does not exist, or if it does exist but the server does not have it.
 * `lgrIdxsInvalid` - Either the request specifies a `ledger_index_max` that is before the `ledger_index_min`, or the server does not have a validated ledger range because it is [not synced with the network](server-doesnt-sync.html).
 
+## Api Version 2
+* - Specifying `ledger_index_min`, `ledger_index_max` and `ledger_index` will result in an invalidParams error
+* - Specifying `ledger_index_min` or `ledger_index_max` out of a valid range will result in lgrIdxMalformed.
+
+Older Api versions can be used. Documented [here][api-version-history]
```
Comment (thejohnfreeman) [inline content/references/http-websocket-apis/public-api-methods/account-methods/account_tx.md]:
```suggestion
Older [API versions][api-version-history] can be used.
```

File: content/references/http-websocket-apis/public-api-methods/ledger-methods/ledger_entry.md [hunk 3]
```diff
@@ -731,7 +731,10 @@ An example of a successful response:
 * `malformedRequest` - The request provided an invalid combination of fields, or provided the wrong type for one or more fields.
 * `unknownOption` - The fields provided in the request did not match any of the expected request formats.
 
+## Api Version 2
+* - When one or more fields specified incorrectly or missing, it will result in an invalidParams error
 
+Older Api versions can be used. Documented [here][api-version-history]
```
Comment (thejohnfreeman) [inline content/references/http-websocket-apis/public-api-methods/ledger-methods/ledger_entry.md]:
```suggestion
Older [API versions][api-version-history] can be used.
```

**Review by thejohnfreeman — CHANGES_REQUESTED** [review]
Links to "here" hide the destination and are hostile to screen readers. Prefer [linking](https://www.smashingmagazine.com/2012/06/links-should-never-say-click-here/) to nouns at the end of a sentence.

**Comment by PeterChen13579** [comment]
@intelliot @thejohnfreeman Thank you for the feedback, that makes complete sense! 😄 I changed it to what @thejohnfreeman suggested~

**Comment by thejohnfreeman** [comment]
I think this will need a style review from one of our writers. My thoughts and opinions:

- "API" needs to be fully capitalized everywhere it is used. Never "Api".
- There should not be a section named "API Version 2". We should not be adding new sections for every API version. I anticipate, especially now that the gates are open, that we will be adding many versions. Alternatively, we can have a `CHANGELOG` section that lists the changes in each API version, or we can add notes to existing sections, e.g. "Possible Errors", tagged with "New in version 2" or "Added in version 2" or "Changed in version 2" or similar.
- We already have a section on xrpl.org explaining [API Versioning](https://xrpl.org/request-formatting.html#api-versioning), and we already document [how to choose the API version in a request](https://xrpl.org/request-formatting.html#websocket-format). We should link the CHANGELOG on GitHub from that page, and we should link the API Versioning section from every "Added in version X" note in the rest of the documentation.

**Comment by PeterChen13579** [comment]
Hi, just a heads up, my internship with Ripple will be ending next Friday. I don't know if there will be a style guideline by then for API version 2, so perhaps @mDuo13 can take over or assign my few changes to someone else. Thanks

**Comment by mDuo13** [comment]
I plan to take a different approach to document API v2 differences before it becomes enabled (currently expected in server version 1.13.0)

**Pull request closed without merging**